Possible PacketBuffer reference count issues with fdbased tun interface and UDP

### Description

I'm working to hunt down a crash within an open-source tool (https://github.com/nicocha30/ligolo-ng) that utilizes gvisor, and the tool crashes with panic upon processing a UDP packet because the call to `pkt.Network()` fails to Pull Up. There are no issues with TCP, and most of the code flow is identical. The tool uses an `fdbased` tun interface to forward traffic on Linux. 

I switched the tool to use `gvisor.dev/gvisor v0.0.0-20230126004520-51ddacb795ea`, instead of a few months old fork of gvisor, the crash exists in both cases.

The tool registers a custom UDP forwarder function, and a response is handled elsewhere.

The crash occurs when the `CreateEndpoint` call is made on the forwarder request object later.

If I add the following code to the forwarder handler function, there is no panic.
```
	ptr_struct := unsafe.Pointer(request)
	ptr_struct = unsafe.Pointer(uintptr(ptr_struct) + uintptr(56))
	ptrTobuf := (*stack.PacketBufferPtr)(ptr_struct)
	ptrTobuf.IncRef()
		
```
I'm sorry if providing context is hard; this is my first issue, and I am not the author of the other tool. Could there be an extra `DecRef()` call within the tun/fdbased code in gvisor that is causing this behavior? The tool worked fine before the bufferv2 changes.


This is what the panic trace looks like:

```
panic: PullUp failed        

goroutine 50 [running]:
gvisor.dev/gvisor/pkg/tcpip/stack.PacketBufferPtr.headerView({0xc000333b40?}, 0x8b41a5?)
        /root/go/pkg/mod/gvisor.dev/gvisor@v0.0.0-20230126004520-51ddacb795ea/pkg/tcpip/stack/packet_buffer.go:369 +0x145
gvisor.dev/gvisor/pkg/tcpip/stack.PacketHeader.Slice({{0xc000002c80?}, 0x400000?})
        /root/go/pkg/mod/gvisor.dev/gvisor@v0.0.0-20230126004520-51ddacb795ea/pkg/tcpip/stack/packet_buffer.go:514 +0x25
gvisor.dev/gvisor/pkg/tcpip/stack.PacketBufferPtr.Network({0xc0000a1800?})
        /root/go/pkg/mod/gvisor.dev/gvisor@v0.0.0-20230126004520-51ddacb795ea/pkg/tcpip/stack/packet_buffer.go:418 +0x5d
gvisor.dev/gvisor/pkg/tcpip/transport/udp.(*ForwarderRequest).CreateEndpoint(0xc0000fd580, 0x4?)
        /root/go/pkg/mod/gvisor.dev/gvisor@v0.0.0-20230126004520-51ddacb795ea/pkg/tcpip/transport/udp/forwarder.go:77 +0xdc
ligolo-ng/pkg/proxy/netstack.HandlePacket.func1()
        /ligolo_custom/build/pkg/proxy/netstack/handlers.go:142 +0x45b
created by ligolo-ng/pkg/proxy/netstack.HandlePacket       
        /ligolo_custom/build/pkg/proxy/netstack/handlers.go
:129 +0x885  
```


### Steps to reproduce

_No response_

### runsc version

_No response_

### docker version (if using docker)

_No response_

### uname

Linux adnan-khan-ligolo-cmm-test 5.10.0-19-cloud-amd64 #1 SMP Debian 5.10.149-2 (2022-10-21) x86_64 GNU/Linux

### kubectl (if using Kubernetes)

_No response_

### repo state (if built from source)

_No response_

### runsc debug logs (if available)

_No response_

--- a/pkg/tcpip/transport/udp/forwarder.go
+++ b/pkg/tcpip/transport/udp/forwarder.go
@@ -47,7 +47,7 @@ func (f *Forwarder) HandlePacket(id stack.TransportEndpointID, pkt stack.PacketB
 	f.handler(&ForwarderRequest{
 		stack: f.stack,
 		id:    id,
-		pkt:   pkt,
+		pkt:   pkt.IncRef(),
 	})
 
 	return true

